### PR TITLE
feat: use orb anchored DID

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by sandrask — APPROVED** [review]

**Review by rolsonquadras — APPROVED** [review]

File: cmd/wallet-js-sdk/src/user/wallet-user.js [hunk 1]
```diff
@@ -255,11 +256,36 @@ export class WalletUser {
...
    */
   async getPreferences(auth) {
-    return await this.wallet.get({
+    let result = await this.wallet.get({
       auth,
       contentType: contentTypes.METADATA,
       contentID: `${METADATA_PREFIX}${this.user}`,
     });
+
+    console.info("check controller "+result.content.controller);
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply cmd/wallet-js-sdk/src/user/wallet-user.js:265]: will change it to debug later

File: cmd/wallet-js-sdk/src/user/wallet-user.js [hunk 2]
```diff
@@ -255,11 +256,36 @@ export class WalletUser {
...
+
+    if (result.content.controller.includes("did:orb:https")){
+      let resolveDID =  this.didManager.resolveOrbDID(result.content.controller)
+      if (resolveDID.didDocumentMetadata && resolveDID.didDocumentMetadata.method) {
+        console.info("check DID if it is published "+resolveDID.didDocumentMetadata.method.published);
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply cmd/wallet-js-sdk/src/user/wallet-user.js:270]: will change it to debug later

File: cmd/wallet-js-sdk/src/user/wallet-user.js [hunk 3]
```diff
@@ -255,11 +256,36 @@ export class WalletUser {
...
+            contentID: `${METADATA_PREFIX}${this.user}`,
+            contentType: contentTypes.METADATA,
+          });
+
+          result.content.controller = resolveDID.didDocumentMetadata.canonicalId
+
+          console.info("did is published will use canonicalId " + result.content.controller);
+
+          await this.saveMetadata(auth, result.content);
+
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply cmd/wallet-js-sdk/src/user/wallet-user.js:283]: updatePreferences call getPreferences

File: cmd/wallet-js-sdk/src/user/wallet-user.js [hunk 4]
```diff
@@ -255,11 +256,36 @@ export class WalletUser {
...
-    return await this.wallet.get({
+    let result = await this.wallet.get({
       auth,
       contentType: contentTypes.METADATA,
       contentID: `${METADATA_PREFIX}${this.user}`,
     });
+
+    console.info("check controller "+result.content.controller);
+
+    if (result.content.controller.includes("did:orb:https")){
```
[the remark this answers is not in the document]
Comment (fqutishat) [reply cmd/wallet-js-sdk/src/user/wallet-user.js]: done

File: cmd/wallet-js-sdk/src/user/wallet-user.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sudeshrshetty) [reply cmd/wallet-js-sdk/src/user/wallet-user.js:283]: ok, got it

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**